### PR TITLE
🗂 submit to collections

### PR DESCRIPTION
**Pull request by stevejpurves**
reworking of the submission process to work with collections, including a new `--collection` cli option.

### CODE REVIEW

File: packages/curvenote-cli/src/submissions/submit.utils.ts [hunk 1]
```diff
@@ -61,41 +87,123 @@ export function venueQuestion(session: ISession) {
...
+  let kindId;
   if (opts?.kind) {
-    const match = kinds.items.find(
+    const match = kinds.find(
       ({ name }: { name: string }) => name.toLowerCase() === opts.kind?.toLowerCase(),
     );
     if (!match) {
       session.log.info(
-        `${chalk.red(`🚨 submission kind "${opts.kind}" is not accepted at venue ${venue}`)}`,
+        `${chalk.bold.red(`⛔️ submission kind "${opts.kind}" is not accepted in the collection "${collectionMoniker(collection)}"`)}`,
```
Comment (fwkoch) [inline packages/curvenote-cli/src/submissions/submit.utils.ts:185]: At some point we will probably want to massage this logging so it covers the UX for both (1) site with collections and (2) site with only a default collection but no collections as far as the user is concerned.

I think less nuance is ok for now though...

File: packages/curvenote-cli/src/submissions/submit.utils.ts [hunk 2]
```diff
@@ -61,41 +87,123 @@ export function venueQuestion(session: ISession) {
...
-  } catch (err: any) {
+  const openCollections = collections.items.filter((c) => c.open);
+  if (openCollections.length === 0) {
+    session.log.info(
+      `${chalk.red(`⛔️ no collections are open for submissions at venue "${venue}"`)}`,
+    );
+    process.exit(1);
+  }
+
+  session.log.debug(`Explicit collection provided: ${opts?.collection}`);
```
Comment (fwkoch) [inline packages/curvenote-cli/src/submissions/submit.utils.ts]: Minor stuff, but this should probably only be logged this way if `opts.collection` is defined.

**Review by fwkoch — APPROVED** [review]
Looking good - my only comments are pretty trivial logging stuff.

I'll put this through the wringer a bit more as I combine it with my work removing `transfer.yml` in favour of `project.id`.

File: packages/curvenote-cli/src/submissions/submit.utils.ts (hunk 1 above)
Comment (stevejpurves) [reply packages/curvenote-cli/src/submissions/submit.utils.ts:185]: yeah the messaging UX overall probably could be improved

**Pull request merged**